Готовая первая проектная работа

sergey-che09 opened the pull request.

sergey-che09 committed:
Работа №1 "С чистого листа"

sergey-che09 committed:
переименовал папку slyles

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,194 @@
+      </div>
+      <ul class="header__list">
+        <li>
+          <a class="header__list-btn content-section_theme_dark" href="#"
```
`content-section_theme_dark` не следует применять для ссылки, сущность не является секцией

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,194 @@
+    <main class="main content-section content-section_theme_dark">
+      <name class="name">
+        <div class="name__text">
+          <h1 class="cover name__left">C</h1>
```
Тег h1 может быть только один на странице. Этот элемент нужно исправить на заголовок меньшего уровня в соответствии со структурой страницы. 
 Спецификация не запрещает использование нескольких h1 на странице, однако данный кейс не является лучшим решением.

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,194 @@
+    <link rel="stylesheet" href="fonts/fonts.css" />
+    <link rel="stylesheet" href="styles/style.css" />
+    <title>С чистого листа</title>
+  </head>
```
Верно оформлен head

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,194 @@
+          <p class="content-description__text">
+            8 способов борьбы со страхом чистого листа
+          </p>
+          <p class="content-description__date">05.06.2022 г.</p>
```
Для даты лучше использовать https://doka.guide/html/time/

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,194 @@
+      </ul>
+    </header>
+    <main class="main content-section content-section_theme_dark">
+      <name class="name">
```
Все Error валидатора https://validator.w3.org/#validate_by_input

AliaksandrHrychonak replied:
Данный раздел является секцией и для его разметки следует использовать семантический тег `<section>`. Это важно для правильной структуры документа и его доступности.

Тег `<section>` представляет собой тематическую группировку контента, обычно с заголовком(*но не обязательно*). Его использование помогает разделить содержимое страницы на логические части, что улучшает семантическую структуру документа и делает его более понятным как для поисковых систем, так и для инструментов доступности.

Например в отличие от универсального тега `<div>`, тег `<section>` имеет семантическое значение и указывает на то, что содержимое внутри него представляет собой отдельный раздел документа. Это особенно важно для скринридеров и других вспомогательных технологий, которые используются людьми с ограниченными возможностями.

Подробнее про тег `<section>` ⇒ https://doka.guide/html/section/

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,194 @@
+          </p>
+        </div>
+        <div class="reasons content-section_theme_bright">
+          <div class="reasons__header">
```
Надо исправить `h2` следует использовать как обёртку вложенного текста, две строки текста представляют заголовок второго уровня

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,194 @@
+            состояние потока. Причин, почему первые усилия даются так сложно,
+            может быть очень много. Вот основные:
+          </p>
+          <ul class="reasons__list">
```
Отличное решение использовать список

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,194 @@
+          </ul>
+        </div>
+        <div class="how-to content-section_theme_dark">
+          <h2 class="reasons__header how-to_right">Как всё-таки начать</h2>
```
Надо исправить `h2` следует использовать как обёртку вложенного текста, две строки текста представляют заголовок второго уровня

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,194 @@
+            происходит.
+          </p>
+        </div>
+        <div class="reasons content-section_theme_bright">
```
Данный раздел является секцией и для его разметки следует использовать семантический тег `<section>`. Это важно для правильной структуры документа и его доступности.

Тег `<section>` представляет собой тематическую группировку контента, обычно с заголовком(*но не обязательно*). Его использование помогает разделить содержимое страницы на логические части, что улучшает семантическую структуру документа и делает его более понятным как для поисковых систем, так и для инструментов доступности.

Например в отличие от универсального тега `<div>`, тег `<section>` имеет семантическое значение и указывает на то, что содержимое внутри него представляет собой отдельный раздел документа. Это особенно важно для скринридеров и других вспомогательных технологий, которые используются людьми с ограниченными возможностями.

Подробнее про тег `<section>` ⇒ https://doka.guide/html/section/

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,194 @@
+            </li>
+          </ul>
+        </div>
+        <div class="how-to content-section_theme_dark">
```
Данный раздел является секцией и для его разметки следует использовать семантический тег `<section>`. Это важно для правильной структуры документа и его доступности.

Тег `<section>` представляет собой тематическую группировку контента, обычно с заголовком(*но не обязательно*). Его использование помогает разделить содержимое страницы на логические части, что улучшает семантическую структуру документа и делает его более понятным как для поисковых систем, так и для инструментов доступности.

Например в отличие от универсального тега `<div>`, тег `<section>` имеет семантическое значение и указывает на то, что содержимое внутри него представляет собой отдельный раздел документа. Это особенно важно для скринридеров и других вспомогательных технологий, которые используются людьми с ограниченными возможностями.

Подробнее про тег `<section>` ⇒ https://doka.guide/html/section/

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,194 @@
+        </div>
+        <div class="how-to content-section_theme_dark">
+          <h2 class="reasons__header how-to_right">Как всё-таки начать</h2>
+          <h2 class="reasons__header how-to_left">что-то делать</h2>
```
Надо исправить

reasons__header не подходит для данной секции, используйте класс без привязки к типу секции, что бы не возникало противоречий контента и класса

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,194 @@
+      <div class="footer__copy">
+        <span>© VЁRSTKA</span>
+        <ul class="footer__links">
+          <li><a class="content-section_theme_dark" href="#"
```
`content-section_theme_dark` не следует применять для ссылки, сущность не является секцией

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,355 @@
+  gap: 32px;
+}
+
+img {
```
При использовании фиксированного значения для скругления углов (например, `border-radius: 20px`), при масштабировании изображения пропорции скругления будут нарушаться. Изображение может стать больше, а скругление останется прежним.

Чтобы сохранить пропорциональное скругление углов независимо от размера изображения, используйте процентное значение: `border-radius: 50%`. Это значение всегда будет вычисляться относительно текущих размеров элемента.

AliaksandrHrychonak commented on style/style.css:
Файл нужно удалить, он не участвует в работе проекта

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,355 @@
+body {
+  margin: 0;
+  font-family: "Inter", sans-serif;
+  background-image: url(https://code.s3.yandex.net/web-code/images/dotted.svg);
```
В фоновом изображении нет необходимости, используйте цвет фона

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,355 @@
+}
+
+h1 {
+  font-family: "EB Garamond";
```
Во всем проекте для всех текстовых элементов и обёрток используется `font-family: Шрифт`, это является не совсем корректным решением. В том случае, если по какой-либо причине шрифты не загрузятся или будет использоваться браузер, не поддерживающий текущий формат шрифта - следует предусмотреть такой исход и добавить системные шрифты. **Необходимо подправить этот момент во всём коде**, нужно добавлять альтернативные шрифты, если `Шрифт` не получилось найти (загрузить). И обязательно в самом конце указать `sans-serif (без засечек)`, иначе браузер применит какой-нибудь стандартный шрифт, который найдет.

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,355 @@
+header {
+  font-size: 18px;
+  width: 360px;
+  height: 132px;
```
Фиксацию высоту необходимо удалить.

Не рекомендуется задавать фиксированную высоту блокам, так как это может привести к проблемам с отображением при изменении количества контента. При добавлении контента он может выйти за пределы блока, а при удалении может образоваться лишнее пустое пространство.

Высота блока должна определяться его содержимым автоматически. Это обеспечит корректное отображение независимо от количества и размера контента внутри блока.

Такой подход делает отображение более гибким и устойчивым к изменениям в будущем, когда может потребоваться добавить или удалить контент.


Аналогично необходимо подправить по всему коду для текста и блоков

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,355 @@
+
+.name {
+  display: flex;
+  height: 230px;
```
При использовании фиксированного значения свойства `height` для блока могут возникнуть проблемы с отображением контента при увеличении кол-во контента в блоке.

Для декоративного блока следует использовать свойство `min-height`, которое задает минимальную высоту блока, позволяя ему автоматически расширяться при необходимости. Это обеспечивает более гибкое и надежное отображение контента.

Подробнее про свойство `min-height` ⇒ https://developer.mozilla.org/en-US/docs/Web/CSS/min-height

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,355 @@
+  padding-bottom: 3px;
+}
+
+.reasons__header-left- {
```
Классы проекта должны иметь однотипный формат написания

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,355 @@
+  padding-top: 10px;
+  padding-bottom: 3px;
+  font-weight: 400;
+  letter-spacing: 0.4px;
```
Стиля нет в макете

sergey-che09 replied:
Данный стиль используется .content-section_theme_gray h3 
<li class="how-to__card **content-section_theme_gray**">
    <div class="how-to__autor">
        <img src="images/bunin.png" alt="фото Ивана Бунина" />
         <p class="how-to__autor_name">Иван Бунин</p>
     </div>
      <**h3**>Сделать из задачи рутину</h3>

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,355 @@
+.how-to h2 {
+  padding: 0;
+  line-height: 32px;
+  letter-spacing: 0.5px;
```
Стиля нет в макете

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,355 @@
+  padding-top: 22px;
+  /* padding-bottom: 1px; */
+  font-weight: 600;
+  letter-spacing: 0.3px;
```
Стиля нет в макете

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,194 @@
+<!DOCTYPE html>
+<html lang="ru">
+  <head>
```
Нужен метатег viewport https://disk.yandex.com/i/_3VsNuS-yxr2QA

AliaksandrHrychonak reviewed (commented):
Необходимо развернуть общий комментарий ⬇️

Добрый день. Работа выполнена на хорошем уровне!

Отлично:

* Код валидный
* Используются семантически верные теги
* Размечена вся структура

Надо исправить:

* Все комментарии с пометкой "надо исправить"
* Все Error валидатора https://validator.w3.org/#validate_by_input
* Страницу нужно центрировать https://disk.yandex.com/i/S6yHLAquuXCt4g
* Цвет текста не по макету https://disk.yandex.com/i/VFMF4i4R208FxA

Можно лучше:

* Все комментарии "Можно лучше" в коде.

Необходимо исправить все "надо исправить". Исправление "можно лучше" остаются на ваше усмотрение.

sergey-che09 committed:
Delete style directory

sergey-che09 committed:
Исправлены перечисленные недостатки

Все указанные классы как не используемые, используются, как пример класс ".content-section_theme_gray h3", видимо валидатор не понимает вложенности.
Не нашел способа как отказаться от размера в шапке height: 132px; 
Изменение пикселей на проценты в закруглениях, дает овальные скругления, т.к. ширина и высота разные.

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,364 @@
+  padding-left: 20px;
+  padding-top: 22px;
+  font-weight: 600;
+  letter-spacing: 0.3px;
```
Стиля нет в макете

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,364 @@
+.how-to h2 {
+  padding: 0;
+  line-height: 32px;
+  letter-spacing: 0.5px;
```
Стиля нет в макете

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,364 @@
+}
+
+.reasons__header-left {
+  letter-spacing: 0.5px;
```
Стиля нет в макете

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,364 @@
+}
+
+.reasons__header-right {
+  letter-spacing: 0.5px;
```
Стиля нет в макете

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,364 @@
+}
+
+.name__text {
+  height: 221px;
```
Фиксацию высоту необходимо удалить.

Не рекомендуется задавать фиксированную высоту блокам, так как это может привести к проблемам с отображением при изменении количества контента. При добавлении контента он может выйти за пределы блока, а при удалении может образоваться лишнее пустое пространство.

Высота блока должна определяться его содержимым автоматически. Это обеспечит корректное отображение независимо от количества и размера контента внутри блока.

Такой подход делает отображение более гибким и устойчивым к изменениям в будущем, когда может потребоваться добавить или удалить контент.

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,364 @@
+header {
+  font-size: 18px;
+  width: 360px;
+  height: 132px;
```
Фиксацию высоту необходимо удалить.

Не рекомендуется задавать фиксированную высоту блокам, так как это может привести к проблемам с отображением при изменении количества контента. При добавлении контента он может выйти за пределы блока, а при удалении может образоваться лишнее пустое пространство.

Высота блока должна определяться его содержимым автоматически. Это обеспечит корректное отображение независимо от количества и размера контента внутри блока.

Такой подход делает отображение более гибким и устойчивым к изменениям в будущем, когда может потребоваться добавить или удалить контент.

AliaksandrHrychonak commented on styles/style.css:
```diff
@@ -0,0 +1,364 @@
+  padding-top: 10px;
+  padding-bottom: 3px;
+  font-weight: 400;
+  letter-spacing: 0.4px;
```
Стиля нет в макете

AliaksandrHrychonak commented on index.html:
```diff
@@ -0,0 +1,199 @@
+    <link rel="stylesheet" href="fonts/fonts.css" />
+    <link rel="stylesheet" href="styles/style.css" />
+    <title>С чистого листа</title>
+  </head>
```
Нужно исправить все Error валидатора https://validator.w3.org/#validate_by_input

AliaksandrHrychonak commented on fonts/fonts.css:
```diff
@@ -0,0 +1,26 @@
+      url(interregular.ttf) format('truetype');
+    font-weight: 400;
+  }
+  
```
В проекте не хватает жирного шрифта `Inter`

AliaksandrHrychonak reviewed (commented):
Добрый день, отличные исправления, проект принят! В проекте было бы хорошо подправить оставшиеся комментарии в коде. 

Так же обратите внимание, что для прохождения дальнейших ревью, делать merge пулл реквеста следует делать только после полного код-ревью
